hazelcast/src/main/java/com/hazelcast/jet/impl/JetService.java

```diff
@@ -122,11 +123,14 @@ public void init(NodeEngine engine, Properties hzProperties) {
         networking = new Networking(engine, jobExecutionService, config.getInstanceConfig().getFlowControlPeriodMs());
 
         ClientEngine clientEngine = engine.getService(ClientEngineImpl.SERVICE_NAME);
-        ExceptionUtil.registerJetExceptions(clientEngine.getExceptionFactory());
-
+        ClientExceptionFactory clientExceptionFactory = clientEngine.getExceptionFactory();
+        if (clientExceptionFactory != null) {
+            ExceptionUtil.registerJetExceptions(clientExceptionFactory);
+        } else {
+            logger.warning("Jet exceptions are not registered since the client exception factory is not accessible.");
```

Comment:
I think this should be logged at fine level. If `NoOpClientEngine` is in use, then it means user explicitly configured Hazelcast member with advanced network config and no server port for clients, so the intent is to not use clients at all with the cluster.

Reply:
Applied this change.